api/types/constants.go

```diff
@@ -441,6 +444,12 @@ const (
 	// InternalResourceIDLabel is a label used to store an ID to correlate between two resources
 	// A pratical example of this is to create a correlation between a Node Provision Token and the Node that used that token to join the cluster
 	InternalResourceIDLabel = "teleport.internal/resource-id"
+
+	// AlertOnLogin is an internal label that indicates an alert should be displayed to users on login
+	AlertOnLogin = "teleport.internal/alert-on-login"
```

Comment:
Wonder why we want to use labels. We can just create a separate filed in alert resource to handle this instead of using labels for that.

Reply:
Either way works well in this case I think.  In general, my gut feeling is that labels are a bit better than GRPC fields for this "alert describes its target audience" thing that I'm trying, mostly because unknown labels still get transmitted and stored, which is really nice since it means we don't need to bump the resource version in order to safely interact with older clients without potentially losing an alert.  Not a huge deal, but I think its nifty.

My plan is to use the upcoming update/versioning work to experiment with this API design before we start to rely on it too much.  If it ends up feeling too limiting, moving to a more structured way of organizing alerts and mapping them to users will still be pretty easy at that stage.  If it does end up being fairly nice to work with, I think the mental model of "labels define the target" is a bit simpler if everything about the target lives in the labels, rather than having a mix of struct fields and labels.